### PR TITLE
Adding Debug/bisect

### PR DESCRIPTION
**Pull request by lkishline**
Looking for feedback on content - will go work on fixing formatting issues now.

### CODE REVIEW

File: git_flow.rst [hunk 1]
```diff
@@ -191,6 +191,54 @@ This synchronizes their local ``master`` branch with the ``master`` branch of
...
+
+    $ git bisect
+
+What it does:
+This command is a way of quickly finding where a problem was introduced in a particular commit. It does so by performing a “binary” search between two different states of the code and via process of elimination - where the issue was first introduced. 
+
+How to use it:
+First, you will need to find a commit where the issue is not present (where it is working). After doing a search through the git log from the time you know things were working, confirm that this commit  does indeed work by switching your checkout to it instead::
+
+    $ git checkout 'abc1234'
```
Comment (Eric89GXL) [inline git_flow.rst]: you shouldn't need the quotes

**Comment by Eric89GXL** [comment]
LGTM, might want to add something about aborting as well

**Comment by lkishline** [comment]
What is the best way to give credit to the sources I followed? (what format should I use for this?)

**Comment by Eric89GXL** [comment]
You can just add HTML links to the original pages

File: git_flow.rst [hunk 2]
```diff
@@ -191,6 +191,56 @@ This synchronizes their local ``master`` branch with the ``master`` branch of
 the ``upstream`` remote repo, and deletes their local ``fix_branch`` (which is
 no longer needed, since its changes have been merged into the upstream master).
 
+Debugging with Git
```
Comment (drammock) [inline git_flow.rst]: Suggest making this  a level-3 heading (add 3 `#` to start of line).

File: git_flow.rst [hunk 3]
```diff
@@ -191,6 +191,56 @@ This synchronizes their local ``master`` branch with the ``master`` branch of
...
+
+What it does:
+This command is a way of quickly finding where a problem was introduced in a particular commit. It does so by performing a “binary” search between two different states of the code and via process of elimination - where the issue was first introduced. 
+
+How to use it:
+First, you will need to find a commit where the issue is not present (where it is working). After doing a search through the git log from the time you know things were working, confirm that this commit  does indeed work by switching your checkout to it instead::
+
+    $ git checkout abc1234
+
+Then you will need to find a commit where things are not working. You will do this by getting the current commit hash.:
```
Comment (Eric89GXL) [inline git_flow.rst]: This needs to be `::` to render properly (check "view" button above to see how it looks currently)

File: git_flow.rst [hunk 4]
```diff
@@ -191,6 +191,56 @@ This synchronizes their local ``master`` branch with the ``master`` branch of
...
+
+Then, one last time - check if this commit works. And let Git know::
+
+    $ git bisect good
+
+Now, you have your problem commit. Once you have updated the issue, you can get back to a working state by letting git bisect know that you are done::
+
+    $ git bisect reset
+
+This returns you to the original HEAD before git bisect start. And allows you to abort/stop the bisecting if need be.
```
Comment (Eric89GXL) [inline git_flow.rst]: might be worth having double-backticks around the "git bisect start" part so it's rendered as code

File: git_flow.rst (hunk 2 above)
Comment (Eric89GXL) [inline git_flow.rst]: Welcome back to GitHub @drammock we've missed you :)

Comment (drammock) [inline git_flow.rst]: oops, forgot this was rst and not markdown.  This should work:

```
Debugging with Git
~~~~~~~~~~~~~~
```

**Comment by Eric89GXL** [comment]
@lkishline don't forget about this PR, too -- we don't want to leave PRs open too long if we can avoid it. It risks merge conflicts and people (devs and reviewers both) forgetting about what the relevant issues and such are. And if you're waiting for review, don't hesitate to ping relevant reviewers (me and Dan probably), since we won't know if it's ready for another round of review unless you tell us (we are not notified of commits).

...speaking of which, we should really also to the `git flow` docs in the Developers and Maintainers sections that iterations on PRs should occur quickly if possible at both modification and review ends for the above mentioned reasons. Can you add that? And also note in the developers section that devs shouldn't hesitate to ask for reviews after new commits are made to fix the last round of comments.

**Comment by Eric89GXL** [comment]
(of course nobody should feel compelled to respond after hours or anything, but PRs sitting idle for a week or more *AHEM* is getting into dangerous territory)

**Comment by lkishline** [comment]
Oh, my bad. I will clean this up tomorrow and get things squared away.

**Comment by Eric89GXL** [comment]
don't sweat it, @drammock has one that he opened last August...

**Comment by lkishline** [comment]
Well I also want to work on a few more sections and make things clearer for people who have very little background with git and the terminology. As it is now, it assumes a certain level of background knowledge. So I'll clean up what we had originally set out to do, tomorrow. But then I might spend some time expanding and proposing some added sections if that is okay.

**Comment by Eric89GXL** [comment]
Yeah, that's fine. I think for this a couple or a few PRs is better than a single monolithic one, it will ease the review process.

**Comment by lkishline** [comment]
Okay, took me a couple tries to get the html links right - sorry about that. this ' and this ` look very similar...
Any ways - I think this is everything we originally set out to do. So if it looks good to everyone then I think we can close it.

**Comment by Eric89GXL** [comment]
Great, thanks @lkishline

**Pull request merged**